### ISSUE TITLE
"Always on Top"?

### ISSUE THREAD

**Issue by chrismetcalf** [felixhageloh/uebersicht#97]
Any chance we could ever create widgets that actually floated above windows on your desktop?

I use @ttscoff's [loadbar](https://github.com/ttscoff/loadbar) widget, and it'd be awesome to turn it into a 1px bar that would always be visible, even if a window was maximized.

It could maybe be triggered by CSS that went above a certain `z-index` "ceiling" - e.g. > 500 makes it float always on top.

**Comment by TraderStf** [comment]
May be by looking into dashboard possibilities, there is a 'developer' mode where you can drag-out a widget from the board and it floats over the finder.

A setting must be enabled before being able to use it, explanations:
http://macs.about.com/od/usingyourmac/qt/widgetondesktop.htm

Did not try on yosemite.

**Comment by jps3** [comment]
I would second the ability to set a Übersicht widget’s window to be set to always be on top. I had previously been using GeekTool which has such an option, and it is useful for some of the Ü widgets I deploy to building staff for monitoring various things (ie. printers) so they do not get covered up by other windows.
Not sure how that would be implemented, I would guess CSS would not be the place (as that has more to do with the actual widget’s docment window’s contents itself. Perhaps a flag in the widget‘s main *.coffee file? Regardless, it would be a very helpful addition if it can be implemented easily for you.

**Comment by mcallaway** [comment]
+1

**Comment by ZedZhang** [comment]
+1

**Comment by arimathea** [comment]
+1

**Comment by kerbyfc** [comment]
+1

**Comment by apoterenko** [comment]
+1

**Comment by felixhageloh** [comment]
I see this is becoming popular :)

It is tough to implement, because unlike Geek Tool (I am assuming at least), widgets don't live in their own window - they all share one large window containing a WebView. Also, it is technically possible to have Übersicht above all windows while still being able to click on windows below, but it turned out to be quite tricky to do with a WebView.

Anyway, if I get any new ideas or somebody has suggestions I can look into this!

**Comment by chrismetcalf** [comment]
@felixhageloh Perhaps as an alternative a way to assign a momentary shortcut key to bring the whole uber window to the top in some sort of transparent layer, kind of like the old OSX Dashboard

**Comment by premjg** [comment]
@ecnepsnai - Looks like you really hate this feature?

**Comment by ecnepsnai** [comment]
No, I hate it when people write "+1" as comments rather than use reactions... but I just realized that all of those comments were made before reactions were added.

**Comment by CedricRBR** [comment]
@felixhageloh What if you defined two windows, each containing their own WebView ? One could be for the "background" widgets and another one for the "always on top" ones.

**Comment by felixhageloh** [comment]
@CedricRBR the main problem is that when the WebView is overlaying the desktop it becomes unusable. This is because clicks are intercepted by the WebView. Its is possible to have a NSView overlaying the desktop and have clicks on transparent bits pass through down to the desktop. However, with a WebView this does not seem to work (well).

**Comment by Peeja** [comment]
@felixhageloh Could you subclass the WebView to override [`hitTest:`](https://developer.apple.com/documentation/appkit/nsview/1483364-hittest?language=objc), then implement it using [`bitmapImageRepForCachingDisplayInRect`](https://developer.apple.com/documentation/appkit/nsview/1483440-bitmapimagerepforcachingdisplayi?language=objc) / [`cacheDisplayInRect:toBitmapImageRep:`](https://developer.apple.com/documentation/appkit/nsview/1483552-cachedisplayinrect?language=objc) to render the pixel being clicked and check its alpha?

**Comment by felixhageloh** [comment]
@Peeja thanks for the tip! I currently have a solution (not released yet) using a tracking rect. This might be an alternative/better solution

**Comment by Peeja** [comment]
@felixhageloh Cheers, looking forward to it! 🎉

**Comment by nkgm** [comment]
I too, would be very interested in this.

**Comment by prurph** [comment]
Any updates on this? It would be great to be able to show a popup window on top of other windows.

**Comment by andrematzke** [comment]
I got a mini system charts-widgets.. always on top would make that very useful

**Comment by yuchen-lea** [comment]
I'm looking forward to the implementation of the "always on top" feature. Currently, Geektool can provide this function. However, Geektool's webpage editing is not as convenient as Ubersicht's. If Ubersicht supports the "always on top" feature, I will then use Ubersicht to handle all my widgets". That would be great

**Comment by geo242** [comment]
8 years later. I am thinking this isn't happening.
